### PR TITLE
[4.7] Build improvements

### PR DESCRIPTION
**Pull request by SgtSilvio**
https://hivemq.kanbanize.com/ctrl_board/4/cards/4845/details/

### CODE REVIEW

File: src/test/java/com/hivemq/cli/DefaultCLIPropertiesTest.java [hunk 1]
```diff
@@ -26,46 +25,37 @@
...
+    static Path pathToNoProperties;
 
     @BeforeAll
     static void setUp() throws IOException {
-        final ClassLoader classLoader = DefaultCLIPropertiesTest.class.getClassLoader();
         System.setProperty("user.home", pathToTmpDir.toString());
         File mqttDir = new File(pathToTmpDir.toString() + "/.mqtt-cli");
         mqttDir.createNewFile();
-        pathToPropertiesDir = classLoader.getResource("PropertyFiles").getPath();
+        pathToPropertiesDir = DefaultCLIPropertiesTest.class.getResource("/PropertyFiles").getPath();
```
Comment (A-Imal) [inline src/test/java/com/hivemq/cli/DefaultCLIPropertiesTest.java:46]: Should it not be getClass().getResource()?

File: build.gradle.kts [hunk 2]
```diff
@@ -568,47 +560,10 @@ distributions.shadow {
...
-            var text = gradleProperties.readText()
-
-            text = text.replace("(?m)^hivemq-api\\.version=.+".toRegex(), "hivemq-api.version=$version")
-            text = text.replace("(?m)^hivemq-swarm-api\\.version=.+".toRegex(), "hivemq-swarm-api.version=$version")
-
-            gradleProperties.writeText(text)
-        }
-    }
+tasks.register<Sync>("updateSpecs") {
+    from(hivemqOpenApi) { rename { "hivemq-openapi.yaml" } }
```
Comment (A-Imal) [inline build.gradle.kts:564]: Does Gradle still build when these projects are not available?

File: src/test/java/com/hivemq/cli/DefaultCLIPropertiesTest.java (hunk 1 above)
Comment (SgtSilvio) [reply src/test/java/com/hivemq/cli/DefaultCLIPropertiesTest.java:46]: we are in a static context here unfortunately

File: build.gradle.kts (hunk 2 above)
Comment (SgtSilvio) [reply build.gradle.kts:564]: yes, as hivemqOpenApi falls back to the file in the specs folder

**Review by A-Imal — APPROVED** [review]

**Review by LukasHiveMQ — APPROVED** [review]

**Pull request merged**